src/python/zquantum/vqe/utils.py

```diff
@@ -1,64 +1,59 @@
-                                until there are no more parameters.
-        has_rx_layer (bool): whether to include the layer of Rx rotations at the beginning of the 
-            ansatz.
-        elementary (bool): if True, decomposes the U1ex and U2ex gates into elementary gates. Useful
-                           if they are not defined on the device/simulator
+        fermion_generator (openfermion.FermionOperator or 
+            openfermion.InteractionOperator): fermionic generator.
+        fermion_transform (str): The name of the qubit to fermion transformation
+            to use.
 
     Returns:
-        dict: dictionary describing the ansatz.
+        zquantum.core.circuit.Circuit: Circuit corresponding to the exponentiation of the
+            transformed operator. 
     """
+    if transformation not in ["Jordan-Wigner", "Bravyi-Kitaev"]:
+        raise RuntimeError(f"Unrecognized transformation {transformation}")
+
+    # Transform generator to qubits
+    if transformation == "Jordan-Wigner":
+        transformation = jordan_wigner
+    elif transformation == "Bravyi-Kitaev":
+        transformation = bravyi_kitaev
+
+    qubit_generator = transformation(fermion_generator)
+
+    for term in qubit_generator.terms:
+        qubit_generator.terms[term] = float(qubit_generator.terms[term].imag)
+    qubit_generator.compress()
 
-    if n_qubits is None and n_mo is None:
-        raise Exception("n_qubits or n_mo must be provided for all Ansatze")
-    if n_qubits is None:
-        n_qubits = 2 * n_mo
+    # Quantum circuit implementing the excitation operators
+    circuit = exponentiate(qubit_generator)
 
-    if ansatz_type == "singlet UCCSD":
-        if(n_alpha != n_beta):
-            raise RuntimeError('Number of alpha and beta electrons must be equal for a singlet UCCSD ansatz')
-        if spin_ordering != 'interleaved':
-            raise RuntimeError('Only the interleaved spin ordering is implemented for singlet UCCSD')
-        n_params = [ uccsd_singlet_paramsize(n_qubits=n_qubits,
-                                           n_electrons=n_alpha+n_beta) ]
-        ansatz =    {'ansatz_type': 'singlet UCCSD',
-                    'ansatz_module': 'zquantum.vqe.ansatzes.ucc',
-                    'ansatz_func' : 'build_singlet_uccsd_circuit',
-                    'ansatz_kwargs' : {
-                            'n_mo' : n_mo,
-                        'n_electrons' : n_alpha+n_beta,
-                        'transformation' : transformation},
-                     'n_params': n_params}
-        return(ansatz)
+    return circuit
+
+
+def create_layer_of_gates(number_of_qubits: int, gate_name: str) -> Circuit:
+    """
+    Creates a circuit consisting of a single layer of specific gate.
+    """
+    circuit = Circuit()
+    circuit.qubits = [Qubit(i) for i in range(0, number_of_qubits)]
+    circuit.gates = []
+    for i in range(number_of_qubits):
+        circuit.gates.append(Gate(gate_name, [circuit.qubits[i]]))
```

Comment:
Can be replaced by a more readable list comprehension.

Reply:
This code has been moved to core, resolving conversation.